airbyte-cdk/bulk/toolkits/load-low-code/src/main/kotlin/io/airbyte/cdk/load/model/checker/Checker.kt

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.model.checker
+
+import com.fasterxml.jackson.annotation.JsonSubTypes
+import com.fasterxml.jackson.annotation.JsonTypeInfo
+
+/** Base interface for all checker types in declarative destinations. */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
+@JsonSubTypes(JsonSubTypes.Type(value = HttpRequestChecker::class, name = "HttpRequestChecker"))
+sealed interface Checker
```

Comment:
so far the pattern for `sealed` is keeping the implementations in the same file.

Reply:
For my understanding, why do we have this rule?